lib/connectors/mongodb/connector.rb

```diff
@@ -110,7 +110,9 @@ def yield_documents
 
             view = cursor.skip(skip).limit(PAGE_SIZE)
             view.each do |document|
-              yield serialize(document)
+              yield_with_handling_transient_errors do
```

Comment:
I think this block should encompass the full block of the `each do |document|` so we capture any errors that somehow occur in the below math lines (or other logic that eventually gets added there)

Reply:
It might not work cause of `break` statement, gonna try it out, but makes sense!